lib/KSelect/KeenUiSelect.vue

```diff
@@ -1,1207 +0,0 @@
-          >
-            <div
-              v-if="hasSearch"
-
-              class="ui-select-search"
-              @click.stop
-
-              @keydown.space.stop
-            >
-              <input
-                ref="searchInput"
-                v-model="query"
-                autocomplete="off"
-                class="ui-select-search-input"
-
-                type="text"
-
-                :placeholder="searchPlaceholder"
-              >
-
-              <UiIcon class="ui-select-search-icon">
-                <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
-                  <path
-                    d="M9.516 14.016c2.484 0 4.5-2.016 4.5-4.5s-2.016-4.5-4.5-4.5-4.5 2.016-4.5
-                    4.5 2.016 4.5 4.5 4.5zm6 0l4.97
-                    4.97-1.5 1.5-4.97-4.97v-.797l-.28-.282c-1.126.984-2.626 1.547-4.22 1.547-3.61
-                    0-6.516-2.86-6.516-6.47S5.906 3 9.516 3s6.47 2.906 6.47 6.516c0 1.594-.564
-                    3.094-1.548 4.22l.28.28h.798z"
-                  />
-                </svg>
-              </UiIcon>
-
-              <KCircularLoader
-                v-show="loading"
-                class="ui-select-search-progress"
-                :size="20"
-                :stroke="4"
-              />
-            </div>
-
-            <ul
-              ref="optionsList"
-              class="ui-select-options"
-              :style="{ backgroundColor: $themeTokens.surface }"
-            >
-              <KeenUiSelectOption
-                v-for="(option, index) in filteredOptions"
-                ref="options"
-                :key="index"
-                :highlighted="isOptionHighlighted(option)"
-                :keys="keys"
-                :multiple="multiple"
-                :option="option"
-                :selected="isOptionSelected(option)"
-
-                :type="type"
-                @click.native.stop="selectOption(option)"
-
-                @mouseover.native.stop="onMouseover(option)"
-              >
-                <slot
-                  name="option"
-
-                  :highlighted="isOptionHighlighted(option)"
-                  :index="index"
-                  :option="option"
-                  :selected="isOptionSelected(option)"
-                ></slot>
-              </KeenUiSelectOption>
-
-              <div v-show="hasNoResults" class="ui-select-no-results">
-                <slot name="no-results">
-                  {{ noResultsText }}
-                </slot>
-              </div>
-            </ul>
-          </div>
-        </transition>
-      </div>
-
-      <div v-if="hasFeedback" class="ui-select-feedback">
-        <div v-if="showError" class="ui-select-feedback-text">
-          <slot name="error">
-            {{ error }}
-          </slot>
-        </div>
-
-        <div v-else-if="showHelp" class="ui-select-feedback-text">
-          <slot name="help">
-            {{ help }}
```

Comment:
Even tho we are not passing in the help prop, I think it would be good to have it in for the future! Also, error prop requires this piece so would be good to have it I think.

Reply:
Yes, makes sense. :+1: